src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2718,6 +2731,24 @@ void UnitAura::AddStaticApplication(Unit* target, uint8 effMask)
     _staticApplications[target->GetGUID()] |= effMask;
 }
 
+void UnitAura::OnTargetHeartbeat(AuraApplication* aurApp)
+{
+    // Eat/Drinking heartbeat animation; Need to find a way to move this in spell script
+    {
+        SpellSpecificType specific = GetSpellInfo()->GetSpellSpecific();
+        if (specific == SPELL_SPECIFIC_FOOD || specific == SPELL_SPECIFIC_FOOD_AND_DRINK)
+            aurApp->GetTarget()->SendPlaySpellVisual(SPELL_VISUAL_KIT_FOOD);
+        else if (specific == SPELL_SPECIFIC_DRINK)
```

Comment:
you can remove the else while at it. You can play both animations if you eat and drink at the same time

Reply:
@Ovahlord These are mutually exclusive conditions. The spell cannot be both ("food" or "food_and_drink") and "drink". But I calling `OnTargetHeartbeat` for all applied auras. If you eat and drink at the same time `OnTargetHeartbeat` will be called for both auras and both animations will be sent

Reply:
On Classic right now if you both eat and drink at the same time - both visuals are sent (I can hear both the munching and the drinking sounds at the same time).

But as the comment states, eventually this needs to be remade into two generic spellscripts and manually applied on all respective food and drink spells. Then there won't be a way to make them mutually exclusive, and both scripts will execute their Heartbeat hooks and send both visuals together.

Reply:
Current PR code behavior is absolutly identical as if there was a generic spellscript for all food and drink spells with `OnTargetHeartbeat` hook. `UnitAura::OnTargetHeartbeat` will be called for all food and drinks auras applied to your character and if you eat and drink at the same time you get both visual effects. This `else-if` statement only needed to avoid unnecessary check. If the checked aura has `SPELL_SPECIFIC_FOOD` or `SPELL_SPECIFIC_FOOD_AND_DRINK` specific then it cannot have `SPELL_SPECIFIC_DRINK`.

Reply:
Moved to spellscript in 5a8285e